File: lw6/2_Vasilyev_Andrey/Sorter.cpp
#include "Sorter.h"

CSorter::CSorter(const std::vector<int> & data)
	:m_data(data)
{

}

void CSorter::SetData(const std::vector<int> & data)
{
	m_data = data;
}

std::vector<int> CSorter::GetData() const
{
	return m_data;
}

File: lw7/2/Vasilyev_Andrey/lab7/Sorter.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab7
{
    public abstract class CSorter
    {
        public enum Type
        {
            MERGE,
            SHELL
        };

        protected ArrayList m_data;

        public ArrayList Data
        {
            get
            {
                return m_data ?? new ArrayList();
            }
            set
            {
                m_data = value;
            }
        }

        public CSorter(ArrayList data)
        {
            m_data = data;
        }

        public abstract void Sort();
    }
}
